platform/features/layout/src/LayoutController.js

```diff
@@ -156,8 +175,8 @@ define(
             // Watch for changes to the grid size in the model
             $scope.$watch("model.layoutGrid", updateGridSize);
 
-            // Position panes when the model field changes
-            $scope.$watch("model.composition", lookupPanels);
+            // Update composed objects on screen, and position panes
+            $scope.$watch("model.composition", refreshComposition);
```

Comment:
Good catch! Changed it to $watchCollection. Oddly, this was working with $watch (I tested a composition change to ensure it actually gets triggered). Possibly what's happening is that somewhere along the line the .composition attribute of the model gets replaced anyway on update. But $watchCollection is definitely more correct.